### ISSUE TITLE
InternalError uses Debug formatting for Display implementation

### ISSUE THREAD

**Issue by mitskevich** [actix/actix-web#188]
InternalError uses Debug formatting for Display implementation:

https://github.com/actix/actix-web/blob/813d1d6e6653308cf97c88b78fa6395b5981622a/src/error.rs#L599-L606

Is this intentional? Wouldn't it be better to use Display trait and just require `Display` implementation on `T`?

Currently, I'm trying to display user-friendly error messages for 4xx statuses in my JSON API based on actix-web. Errors are returned by handler functions (with `ErrorBadRequest` and similar functions) and it seems that they are being stored in `HttpResponse` as `InternalError`. When I try to format errors for output, I get debug formatting and it seems there is no way to properly format them as user-friendly messages.

**Comment by fafhrd91** [comment]
Maybe typo, I’ll check. 

You can also define your own error types https://actix.rs/book/actix-web/sec-5-errors.html

**Comment by fafhrd91** [comment]
fixed by 2c8d987

**Comment by mitskevich** [comment]
> You can also define your own error types https://actix.rs/book/actix-web/sec-5-errors.html

Yes, but I wouldn't be able to use all helper functions like `actix_web::error::ErrorBadRequest` and the like in my handlers, because they wrap errors inside `InternalError` (and it was displayed with `Debug`). Or I would have had to reimplement `InternalError` and all those functions in my library... 

Anyway, that commit fixes the problem, thanks!

**Comment by fafhrd91** [comment]
if you define custom error you don't need to use `InternalError`. just implement `ResponseError` for your custom error and generate whatever `HttpResponse` you need

**Comment by mitskevich** [comment]
I actually do that for some cases, but I found that sometimes the same error should be returned with different status codes... For example, if I can't parse some value from user's query string it's `400`, but if I can't parse the same type from database record, it's `502`. But it is the same `DeError` in both cases... 

Because of that, my error type would not only have to wrap `DeError`, but also distinguish places where this error can happen... And automatic conversion between `DeError` and my error type wouldn't be possible because of that.

This complicates error handling a bit too much, and I have found that using generic errors, and providing status code information by `actix_web::error::*` helper functions is suitable approach in most cases, so my handlers currently look like this:

```
fn handle_something(req: &HttpRequest) -> Box<Future<Item=ValueType, Error=actix_web::Error>> {  ... }
```

Inside handler's future I can return any error and provide any status code with helper functions.
